### PR TITLE
Fix reload after navigation

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

In the LosingRewardNeuronsModal component, a list of inactive neurons is displayed (see the first screenshot). When a user clicks on a neuron, they should be navigated to the neuron detail page. However, due to stopPropagation being applied to every click inside the Modal component, the SvelteKit router cannot handle internal navigation correctly. Instead, the link is treated as a raw HTML link, resulting in navigation with a full page reload, which is undesirable.

I tried removing stopPropagation from the Modal component, but this caused some undestroyed state issues after navigation. There may have been good reasons for using stopPropagation, but we’ll need more time to investigate.

The svelte/goto method resolves this issue by allowing the router to handle navigation correctly without a page reload. However, it introduces another problem: as I understand it, the open modal prevents Svelte from properly destroying the modal component during navigation, resulting in both pages being rendered simultaneously, breaking the app (see the second screenshot).

To fix this, we need to:
1.	Close the modal.
2.	Navigate using goto.

**TL;DR:** Replace the `<a href...` with goto to avoid page reloads from the modal.

**Note:** This is not a real fix but more of a workaround and likely needs further attention.

---

**The Losing Rewards modal**
image

**Both pages rendered together**
image

# Changes

1. Replace the href with role="button" to keep the card clickable.
2. Make the neuron card dispatch an nnsClick event.
3. Close the modal and use goto to navigate to the neuron details page.

# Tests

- Added.

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/lib/modals/neurons/LosingRewardNeuronsModal.svelte [hunk 1]
```diff
@@ -8,15 +8,30 @@
...
   onMount(() => listKnownNeurons());
 
   const confirm = () => {
     // TBD
   };
+  const close = () => dispatcher("nnsClose");
+
+  const navigateToNeuronDetail = async (neuron: NeuronInfo) => {
+    close();
+    goto(
```
Comment (yhabib) [inline frontend/src/lib/modals/neurons/LosingRewardNeuronsModal.svelte:28]: you could introduce  a tick of svelte. it would allow it to catch up with stuff to do and the do the navigation. 

```suggestion
  await tick();
  goto(buildNeuronUrl({
     universe: OWN_CANISTER_ID_TEXT,
      neuronId: neuron.neuronId,
  })
```

Comment (mstrasinskis) [reply frontend/src/lib/modals/neurons/LosingRewardNeuronsModal.svelte:28]: Thank you for mentioning this. But in this case, the current solution works, and I’d prefer to avoid adding code just for precaution.

File: frontend/src/lib/components/neurons/NnsLosingRewardsNeuronCard.svelte [hunk 2]
```diff
@@ -37,9 +32,10 @@
 
 <Card
   testId="nns-loosing-rewards-neuron-card-component"
-  {href}
+  role="button"
```
Comment (yhabib) [inline frontend/src/lib/components/neurons/NnsLosingRewardsNeuronCard.svelte:35]: This is likely outside the scope of this PR. I believe it would be more appropriate to use:
```
  role="link"
  aria-href="/some-path"
 ```
Additionally, regardless of the role, don't we want to make it keyboard-focusable?
```
  tabindex="0"
```

**Review by yhabib — APPROVED** [review]

File: frontend/src/lib/components/neurons/NnsLosingRewardsNeuronCard.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/neurons/NnsLosingRewardsNeuronCard.svelte:35]: Good point! The Card component should probably include `tabindex="0"` when using the role attribute as button or link. I’ll take a look at it later.
